docs/syntax.md

```diff
@@ -0,0 +1,515 @@
+# syntax
+
+Available commands:
+
+- [`index`](#index)
+- [`select`](#select)
+- [`select into`](#select-into)
+- [`status`](#status)
+- [`compact`](#compact)
+- [`topology`](#topology)
+- [`reindex`](#reindex)
+- [`iterator pop`](#iterator-pop)
+- [`dataset taint`](#dataset-taint)
+- [`dataset untaint`](#dataset-untaint)
+- [`dataset drop`](#dataset-drop)
+- [`config get`](#config-get)
+- [`config set`](#config-set)
+
+All responses from the database use the JSON format.
+
+## index
+
+Examples:
+
+- `index "/opt/something";`
+- `index "/opt/something" "/opt/foobar";`
+- `index "/opt/something" with [gram3, text4, hash4, wide8];`
+- `index from list "/tmp/file-list.txt";`
+- `index "/opt" with taints ["kot"];`
+- `index "/opt" nocheck;`
+
+### basic indexing
+
+A directory can be indexed recursively with `index` command:
+
+```
+index "/opt/something";
+```
+
+Multiple paths can be specified at once:
+
+```
+index "/opt/something" "/opt/foobar";
+```
+
+By default, `gram3` index will be used (this is probably [not what you want](./indextypes.md). Index types may be specified manually:
+
+```
+index "/opt/something" with [gram3, text4, hash4, wide8];
+```
+
+### read a list of files from file
+
+It's also possible to read a file containing a list of targets to be indexed,
+each one separated by a newline.
+
+```
+index from list "/tmp/file-list.txt"
+```
+
+or
+
+```
+index from list "/tmp/file-list.txt" with [gram3, text4, hash4, wide8];
+```
+
+`/tmp/file-list.txt` may contain for example:
+
+```
+/opt/something
+/opt/foobar
+```
+
+### Tag files when indexing
+
+It's possible to tag immediately files during indexing, with a special syntax:
+
+```
+index "/opt" with taints ["kot"]
+```
+
+### Disable safety measures
+
+By default, every new file will be cross-checked with the current state of
+the database, to ensure that no duplicates are added. Users can opt-out
+of this (for performance or other reasons) with `nocheck` modifier.
+
+```
+index "/opt" nocheck
+```
+
+Of course, care should be taken in this case.
+
+### Response format
+
+```
+{
+    "result": {
+        "status": "ok"
+    },
+    "type": "ok"
+}
+```
```

Comment:
What happens if there's an error?

Reply:
I've added an explaination to the top of the document, with generic schemas for success and error results.